data/DataError.go

```diff
@@ -0,0 +1,88 @@
+package data
+
+import "fmt"
+
+//ErrorSet contains a list of Error's
+type ErrorSet struct {
+	errors []*Error
+}
+
+//NewErrorSet creates an initialised ErrorSet
+func NewErrorSet() *ErrorSet {
+	return &ErrorSet{}
+}
+
+//AppendError appends an error to the ErrorSet
+func (e *ErrorSet) AppendError(err *Error) {
+	if err == nil || err.IsEmpty() {
+		return
+	}
+	e.errors = append(e.errors, err)
+	return
+}
+
+//Error returns a string representation of the errors
+func (e *ErrorSet) Error() string {
+
+	errorsStr := ""
+	if e != nil {
```

Comment:
Okay, at first I thought that because the pointer was nil there was no way to call the function.  But, with golang, that's not true (and that's really f*ed up).  Technically, in *every* function for a struct pointer you need to check against the pointer being nil. Frankly, I'm surprised this is the case because I see no reason to allow this, its error prone, and the Golang designers should know better. I'd really like to understand their rationale for supporting this (rather than just panicking if you call a function on a nil pointer (eg. `e.Error()` when e is nil)).